arch/arc/core/timestamp.c

```diff
@@ -29,10 +29,10 @@ u64_t z_tsc_read(void)
 	u64_t t;
 	u32_t count;
 
-	key = irq_lock();
+	key = arch_irq_lock();
```

Comment:
Here I also don't see any reason to have global local (i.e. all IRQs on this core) IRQ disable at all. What are we trying to mitigate here? We don't deal with global variables, so it seems to be pretty re-entrant friendly.

As for theoretical possibility to get an IRQ in between these 2 ops and get incorrect reading... well we may as well get IRQ later and get yet not very precise reading returned.

Reply:
Looks like a valid race condition to protect against to me - don't want tsc_read() go back in time due to a tick interrupt while calling.